parley/src/layout/cursor.rs

```diff
@@ -505,6 +505,16 @@ impl Selection {
         }
     }
 
+    /// Returns a new selection with the focus extended to the given cursor.
+    #[must_use]
+    pub fn extend_to_cursor(&self, focus: Cursor) -> Self {
+        Self {
+            anchor: self.anchor,
+            focus,
+            h_pos: None,
+        }
+    }
+
```

Comment:
Perhaps this is not necessary. I used `Selection` to get the geometry of the preedit to draw a highlight manually, but now the IME preedit is visually distinguished by pushing a ranged style, that's no longer necessary. The preedit text location can be tracked using a simple byte range.

That does still leave the question of how to handle the IME cursor as given by Winit.